src/core/types.ts

```diff
@@ -69,3 +70,8 @@ export abstract class ComplexType extends Type {
         return this.isValidSnapshot(value)
     }
 }
+
+export function typecheck(type: IType, snapshot: any) {
+    if (!type.is(snapshot))
```

Comment:
TODO: Maybe on production build this could be removed to speedup execution, but fails with errors?